### DISCUSSION TITLE
Possibility to start Valetudo update through Homeassistant?

### DISCUSSION THREAD

**Discussion by FaserF** [Hypfer/Valetudo#1890] [Q&A / Support] [answered]
Since quite some time, Homeassistant has added a feature to display new firmware updates within Homeassistant Updates and also to start the updates from there.
This would be pretty neat, since I am not using the Valetudo Web UI at all and do everything from within Homeassistant.
Technically this should be possible I think

https://www.home-assistant.io/integrations/update/
https://developers.home-assistant.io/docs/core/entity/update/

**Comment by Hypfer** [comment, marked as answer]
Hi,

while that initially might sound like a good idea, in reality, it is one of the worst things you can do. In fact, I'm fairly certain that you can kill the project that way.


The main reason why FOSS projects die is not because they don't receive enough funding or because there's not enough time. While those things do also matter, you can make do without them.
The one thing a project cannot do without is appreciation, respect, visibility and seeing the actual humans and their work behind it.

The one thing that keeps people going is seeing that people are thankful, seeing that they provide value and also feeling that they are being valued for their work.


Doing what you suggested nukes that. Completely.
By being this integrated in HA so that the user never even sees your software, you become an implementation detail that can always be swapped out. You as the maintainer become a mere janitor that works upkeeping someone elses property.

You become the random person in Nebraska https://xkcd.com/2347/


I will **never** do that to myself because I **expect** people to know that there is **a real person** behind Valetudo. That it doesn't just exist because of a faceless collective of magic elfs just made it appear out of thin air.

It's hard work. It's my work.
**I won't marginalize myself.**


The whole idea behind the updater entity really only makes sense for commercial products where the user has already paid something and thus the whole thing doesn't rely on being visible. It was conceived for that use-case. It works with that use-case.

It does **not** work with the FOSS use-case.
I am aware that there are quite a few FOSS projects that fell into that trap but just because others were tricked into self-exploitation doesn't mean that I have to follow them down that bottomless pit.

Besides, I think users also have an interest in Valetudo staying the way it is.
Maintained and all and with a Maintainer that isn't burned out to a crisp.

____

You know those happenings when the internet is outraged because a FOSS maintainer has done something unthinkable like adding monetization or tracking or whatever? That's usually the exact result of this self-marginalization and lack of appreciation by the users.
Unfortunately, people usually only double-down on the poor soul, talking them even further down, creating hostile forks etc.
It's a mess and it needs to change. This is what is wrong with FOSS. We must not treat it the same way as we treat commercial software because its inner workings are **entirely** different.

The only reason why it even sorta works at all is that it's unsustainably burning through motivated and naive human resources.
Unfortunately, that burn rate seems to be higher than what the earth can sustain. And beside that, it's also highly inhumane..

____

Here are some additional misc notes:

Apart from this giant conceptual issue, the user also won't read any changelogs that way which is terrible because that just means more support issues for the maintainer and absolutely no visibility of the work being done.

And on top of that, the way the updater entity is implemented pretty much expects that there's a cronjob that periodically pings the cloud checking for new updates. Something that has no business being part of Valetudo.
